addons/mail/models/mail_thread.py

```diff
@@ -4141,15 +4141,14 @@ def _get_mail_thread_data(self, request_list):
         if 'attachments' in request_list:
             res['attachments'] = self._get_mail_thread_data_attachments()._attachment_format()
         if 'followers' in request_list:
-            res['followers'] = [{
-                'id': follower.id,
-                'partner_id': follower.partner_id.id,
-                'name': follower.name,
-                'display_name': follower.display_name,
-                'email': follower.email,
-                'is_active': follower.is_active,
-                'partner': follower.partner_id.mail_partner_format()[follower.partner_id],
-            } for follower in self.message_follower_ids]
+            domain = [
+                ("res_id", "=", int(self.id)),
+                ("res_model", "=", self._name),
+            ]
+            self_domain = domain + [['partner_id', '=', self.env.user.partner_id.id]]
+            res['followersCount'] = len(self.message_follower_ids)
+            res['selfFollower'] = self.env['mail.followers'].search(self_domain)._follower_format()[0] if self.message_is_follower else False
```

Comment:
message_is_follower is already doing a search_fetch, so not sure you can any query by doing the if ? In all cases yo will do the query in the end (two if follower, one if not follower). Could you check ?